instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_2/JettyClientHttpAttributesExtractor.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_2;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpAttributesExtractor;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JettyClientHttpAttributesExtractor extends HttpAttributesExtractor<Request, Response> {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(JettyClientHttpAttributesExtractor.class);
+
+  private static final String[] httpFlavors = {"0.9", "1.0", "1.1", "2.0"};
+
+  @Override
+  @Nullable
+  protected String method(Request request) {
+    return request.getMethod();
+  }
+
+  @Override
+  @Nullable
+  protected String url(Request request) {
+    return request.getURI().toString();
+  }
+
+  @Override
+  @Nullable
+  protected String target(Request request) {
+    String queryString = request.getQuery();
+    return queryString != null ? request.getPath() + "?" + queryString : request.getPath();
+  }
+
+  @Override
+  @Nullable
+  protected String host(Request request) {
+    return request != null ? request.getHost() : null;
+  }
+
+  @Override
+  @Nullable
+  protected String route(Request request) {
+    return null;
+  }
+
+  @Override
+  @Nullable
+  protected String scheme(Request request) {
+    return request != null ? request.getScheme() : null;
+  }
+
+  @Override
+  @Nullable
+  protected String userAgent(Request request) {
+    if (request != null) {
+      HttpField agentField = request.getHeaders().getField(HttpHeader.USER_AGENT);
+      return agentField != null ? agentField.getValue() : null;
+    }
+    return null;
+  }
+
+  @Override
+  @Nullable
+  protected Long requestContentLength(Request request, @Nullable Response response) {
+    Long reqContentLength = null;
+    if (request != null) {
+      HttpField requestContentLengthField =
+          request.getHeaders().getField(HttpHeader.CONTENT_LENGTH);
+      reqContentLength = getLongFromJettyHttpField(requestContentLengthField);
+    }
+    return reqContentLength;
+  }
+
+  @Override
+  @Nullable
+  protected Long requestContentLengthUncompressed(Request request, @Nullable Response response) {
+    return null;
+  }
+
+  @Override
+  @Nullable
+  protected String flavor(Request request, @Nullable Response response) {
+
+    if (response != null) {
+      HttpVersion httpVersion = response.getVersion();
```

Comment:
I was trying to to use the Enum directly and work off Strings with `endsWith()`, because in Jetty10 the values seemed to shift.  But I in the Jetty10 client yet to come, I can use the Jett10 enum

Reply:
Code change here needed to have special handling for HTTP_2.0 enum field, that enum field is different between Jetty 9.2 and the end of the jetty9 series 9.4.